static/app/views/performance/traceDetails/newTraceDetailsValueRenderer.tsx

```diff
@@ -0,0 +1,55 @@
+import type React from 'react';
+import styled from '@emotion/styled';
+
+import {space} from 'sentry/styles/space';
+
+function renderObject(obj: object): React.ReactNode {
+  if (Array.isArray(obj)) {
+    return (
+      <ListContainer>
+        {obj.map((x, i) => (
+          <li key={'array-' + i}>{renderSpanDetailsValue(x)}</li>
+        ))}
+      </ListContainer>
+    );
+  }
+  return (
+    <ObjectContainer>
+      {Object.keys(obj).map(key => (
+        <div key={key}>
+          {key}: {renderSpanDetailsValue(obj[key])}
+        </div>
+      ))}
+    </ObjectContainer>
+  );
+}
+
+export function renderSpanDetailsValue(value: any): React.ReactNode {
+  if (typeof value === 'string') {
+    return value;
+  }
+  if (typeof value === 'object') {
+    return renderObject(value);
+  }
+  return JSON.stringify(value, null, 4);
```

Comment:
do we need to try/catch this?

Reply:
wasn't try/caught before, it's probably fine not to